numpy/core/src/multiarray/item_selection.c

```diff
@@ -1104,13 +1104,10 @@ PyArray_Sort(PyArrayObject *op, int axis, NPY_SORTKIND which)
     int axis_orig = axis;
     int  n = PyArray_NDIM(op);
 
-    if (axis < 0) {
-        axis += n;
-    }
-    if (axis < 0 || axis >= n) {
-        PyErr_Format(PyExc_ValueError, "axis(=%d) out of bounds", axis_orig);
-        return -1;
+    if (check_and_adjust_axis(&axis, n) < 0) {
+        return NULL;
```

Comment:
Same here: I think the return argument shouldn't change, i.e., return `-1` here?